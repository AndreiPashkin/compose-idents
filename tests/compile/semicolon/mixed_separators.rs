//! Using mixed separator styles (semicolon and comma) should result in a compile-time error.
use compose_idents::compose_idents;

compose_idents!(
    my_fn_1 = concat(foo, _, "baz"),
    my_fn_2 = concat(spam, _, eggs); {
    fn my_fn_1() -> u32 {
        123
    }

    fn my_fn_2() -> u32 {
        321
    }
});

fn main() {
    assert_eq!(foo_baz(), 123);
    assert_eq!(spam_eggs(), 321);
}
