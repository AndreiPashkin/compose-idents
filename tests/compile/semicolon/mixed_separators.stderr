error: Mixing "," and ";" as separators is not allowed
 --> tests/compile/semicolon/mixed_separators.rs:6:38
  |
6 |     my_fn_2 = concat(spam, _, eggs); {
  |                                      ^

error[E0425]: cannot find function `foo_bar` in this scope
  --> tests/compile/semicolon/mixed_separators.rs:17:16
   |
17 |     assert_eq!(foo_bar(), 123);
   |                ^^^^^^^ not found in this scope

error[E0425]: cannot find function `spam_eggs` in this scope
  --> tests/compile/semicolon/mixed_separators.rs:18:16
   |
18 |     assert_eq!(spam_eggs(), 321);
   |                ^^^^^^^^^ not found in this scope
